src/overlays/actors/ovl_Door_Spiral/z_door_spiral.c

```diff
@@ -31,30 +40,221 @@ const ActorInit Door_Spiral_InitVars = {
+
+SpiralInfo sSpiralInfoTable[] = {
+    { NULL, NULL, 0, 130, 12, 50, 15 },
+    { gameplay_dangeon_keep_DL_0219E0, gameplay_dangeon_keep_DL_01D980, 0, 130, 12, 50, 15 },
+    { object_numa_obj_DL_004448, object_numa_obj_DL_0007A8, 0, 130, 12, 50, 15 },
+    { object_numa_obj_DL_0051B8, object_numa_obj_DL_0014C8, 0, 130, 12, 50, 15 },
+    { object_hakugin_obj_DL_009278, object_hakugin_obj_DL_006128, 0, 130, 12, 50, 15 },
+    { object_ikana_obj_DL_013EA8, object_ikana_obj_DL_012B70, 0, 130, 12, 50, 15 },
+    { object_ikninside_obj_DL_000EA0, object_ikninside_obj_DL_000590, 0, 130, 12, 50, 15 },
+    { object_danpei_object_DL_002110, object_danpei_object_DL_0012C0, 0, 130, 12, 50, 15 },
+};
+
+typedef struct {
+    /* 0x0 */ s16 objectId;
+    /* 0x2 */ u8 index;
+} SpiralObjectInfo; // size = 0x4
+
+SpiralObjectInfo sSpiralObjectInfoTable[] = {
+    { GAMEPLAY_KEEP, 0 },    { GAMEPLAY_DANGEON_KEEP, 1 }, { OBJECT_NUMA_OBJ, 2 },      { OBJECT_HAKUGIN_OBJ, 4 },
+    { OBJECT_IKANA_OBJ, 5 }, { OBJECT_DANPEI_OBJECT, 7 },  { OBJECT_IKNINSIDE_OBJ, 6 },
+};
+
+typedef struct {
+    /* 0x0 */ s16 sceneId;
+    /* 0x2 */ u8 index;
+} SpiralSceneInfo; // size = 0x4
+
+SpiralSceneInfo sSpiralSceneInfoTable[] = {
+    { SCENE_MITURIN, 2 },     { SCENE_HAKUGIN, 3 },   { SCENE_INISIE_N, 4 }, { SCENE_INISIE_R, 4 },
+    { SCENE_DANPEI2TEST, 5 }, { SCENE_IKNINSIDE, 6 }, { SCENE_CASTLE, 6 },
 };
-*/
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Door_Spiral/func_809A2B60.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F(scale, 1, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 400, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 400, ICHAIN_STOP),
+};
+
+void func_809A2B60(DoorSpiral* this, DoorSpiralActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+    this->unk14A = 0;
+}
+
+s32 func_809A2B70(DoorSpiral* this, PlayState* play) {
+    SpiralObjectInfo* spiralObjectInfo = &sSpiralObjectInfoTable[this->unk147];
+    this->unk148 = spiralObjectInfo->index;
+    if ((this->unk148 == 7) || ((this->unk148 == 2) && (play->roomCtx.curRoom.enablePosLights != 0))) {
+        if (this->unk148 == 2) {
+            this->unk148 = 3;
+        }
+        this->actor.flags |= ACTOR_FLAG_10000000;
+    }
+    func_809A2B60(this, func_809A2FF8);
+    return 0;
+}
+
+//#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Door_Spiral/func_809A2BF8.s")
+u8 func_809A2BF8(PlayState* play) {
+    SpiralSceneInfo* spiralSceneInfo = sSpiralSceneInfoTable;
+    s32 i;
+    u8 var_v1;
+
+    for (i = 0; i < ARRAY_COUNT(sSpiralSceneInfoTable); i++) {
+        if (play->sceneId == spiralSceneInfo->sceneId) {
+            break;
+        }
+        spiralSceneInfo += 0x1;
+    }
+    if (i < ARRAY_COUNT(sSpiralSceneInfoTable)) {
+        var_v1 = spiralSceneInfo->index;
+    } else {
+        var_v1 = (Object_GetIndex(&play->objectCtx, GAMEPLAY_DANGEON_KEEP) >= 0) ? 1 : 0;
```

Comment:
This was recently renamed. You'll need to run namefixer on this file. That's why Jenkins isn't happy